driver-mapping/src/main/java/com/datastax/driver/mapping/PropertyMapper.java

```diff
@@ -15,182 +15,28 @@
-    }
-
-    @SuppressWarnings("unchecked")
-    <A extends Annotation> A annotation(Class<A> annotationClass) {
-        return (A) annotations.get(annotationClass);
-    }
-
-    boolean isComputed() {
-        return hasAnnotation(Computed.class);
-    }
-
-    boolean isTransient() {
-        return hasAnnotation(Transient.class);
-    }
-
-    boolean isPartitionKey() {
-        return hasAnnotation(PartitionKey.class);
-    }
-
-    boolean isClusteringColumn() {
-        return hasAnnotation(ClusteringColumn.class);
-    }
-
-    private String inferColumnName() {
-        if (isComputed()) {
-            return annotation(Computed.class).value();
-        }
-        boolean caseSensitive = false;
-        String columnName = propertyName;
-        if (hasAnnotation(Column.class)) {
-            Column column = annotation(Column.class);
-            caseSensitive = column.caseSensitive();
-            if (!column.name().isEmpty())
-                columnName = column.name();
-        } else if (hasAnnotation(com.datastax.driver.mapping.annotations.Field.class)) {
-            com.datastax.driver.mapping.annotations.Field udtField = annotation(com.datastax.driver.mapping.annotations.Field.class);
-            caseSensitive = udtField.caseSensitive();
-            if (!udtField.name().isEmpty())
-                columnName = udtField.name();
-        }
-        return caseSensitive ? Metadata.quote(columnName) : columnName.toLowerCase();
-    }
-
-    @SuppressWarnings("unchecked")
-    private TypeToken<Object> inferJavaType() {
-        Type type;
-        if (getter != null)
-            type = getter.getGenericReturnType();
-        else
-            type = field.getGenericType();
-        return (TypeToken<Object>) TypeToken.of(type);
-    }
-
-    private int inferPosition() {
-        if (isPartitionKey()) {
-            return annotation(PartitionKey.class).value();
-        }
-        if (isClusteringColumn()) {
-            return annotation(ClusteringColumn.class).value();
-        }
-        return -1;
-    }
-
-    private TypeCodec<Object> createCustomCodec() {
-        Class<? extends TypeCodec<?>> codecClass = getCustomCodecClass();
-        if (codecClass.equals(Defaults.NoCodec.class))
-            return null;
-        @SuppressWarnings("unchecked")
-        TypeCodec<Object> instance = (TypeCodec<Object>) ReflectionUtils.newInstance(codecClass);
-        return instance;
-    }
-
-    private Class<? extends TypeCodec<?>> getCustomCodecClass() {
-        Column column = annotation(Column.class);
-        if (column != null)
-            return column.codec();
-        com.datastax.driver.mapping.annotations.Field udtField = annotation(com.datastax.driver.mapping.annotations.Field.class);
-        if (udtField != null)
-            return udtField.codec();
-        return Defaults.NoCodec.class;
-    }
-
-    @Override
-    public String toString() {
-        return propertyName;
-    }
+public interface PropertyMapper {
```

Comment:
> could there be another kind of implementation?

There used to be a test that completely bypassed reflection in favor of hard-coded casting (in case the mapped classes are known in advance), but I can't find it anymore.

> Also I don't understand why we expose those 2 methods considering their default (and only) implementation will call the same method internally

They call the same method but the second argument (the annotations allowed on the properties) is different.